src/pages/task-detail-page/index.tsx

```diff
@@ -0,0 +1,41 @@
+import { useSelector } from "react-redux";
+import { Link } from "react-router-dom";
+import { RootState } from "../../store/store";
+import { useParams } from "react-router";
+import TaskDetail from "../../components/task-detail/TaskDetail";
+import { ToDoItem } from "../../interfaces/interfaces";
+
+type NumberParams = {
+  id: string;
+};
+
+function TaskDetailPage() {
+  const list = useSelector((state: RootState) => state.list.list);
+  const { id } = useParams<NumberParams>();
+  const idNumber = parseInt(id);
+  const selectedItem = list.length !== 0 ? list.filter((item: ToDoItem) => item.id === idNumber)[0] : "";
```

Comment:
Should be checking length of the filtered list, not the base list.
i.e.
`const filteredList = list.filter((item: ToDoItem) => item.id === idNumber)l`
`const item = list.length > 0 ? filteredList[0] : null;`